etl/steps/data/garden/demography/2023-07-03/population_doubling_times.py

```diff
@@ -0,0 +1,103 @@
+"""Load a meadow dataset and create a garden dataset.
+
+This dataset captures the number of years it takes for a region to double its population size.
+
+To do so, we use our OMM population indicator. We linearly interpolate its values, so that we have a complete time-series of population.
+
+NOTE 1: For now, this step and its Grapher counterpart are only capturing this data for the World and ignore the remaining regions.
+NOTE 2: In the future, we might want to have other countries and regions in this dataset. In that scenario, please review all the code below and Grapher's.
+
+"""
+
+from typing import cast
+
+import pandas as pd
+from owid.catalog import Dataset, Table
+from structlog import get_logger
+
+from etl.helpers import PathFinder, create_dataset
+
+# Get paths and naming conventions for current step.
+paths = PathFinder(__file__)
+log = get_logger()
+
+
+def run(dest_dir: str) -> None:
+    log.info("population_doubling_times: start")
+    #
+    # Load inputs.
+    #
+    # Load meadow dataset.
+    log.info("population_doubling_times: load data")
+    ds_meadow = cast(Dataset, paths.load_dependency("population"))
+
+    # Read table from meadow dataset.
+    tb = ds_meadow["population"]
+    tb.metadata.short_name = "population_doubling_times"
+
+    #
+    # Process data.
+    #
+    # Keep only World data, reset index
+    tb = tb.reset_index()
+    tb = tb[tb["country"] == "World"][["year", "population"]]
+
+    # Linearly interpolate population
+    log.info("population_doubling_times: interpolate values (missing years)")
+    # Create new index ([start_year, end_year])
+    idx = pd.RangeIndex(tb.year.min(), tb.year.max() + 1, name="year")
+    tb = tb.set_index(["year"]).reindex(idx)
+    # Interpolate population values
+    tb["population"] = tb["population"].interpolate(method="index")
+    tb = tb.reset_index()
+
+    # Estimate number of years passed since population was half
+    def _get_year_half_population(population: int, tb: Table):
+        tb_ = tb.loc[(tb["population"] <= population)]
+        idx = (tb_["population"] - population / 2).abs().argmin()
+        year = tb_.iloc[idx]["year"]
+        return year
+
+    log.info("population_doubling_times: estimate years since half population")
+    year_with_half_population = tb["population"].apply(lambda x: _get_year_half_population(x, tb))
+    year_with_half_population.metadata.unit = "years"
+    tb["year"].metadata.unit = "years"
+    tb["years_since_half_population"] = tb["year"] - year_with_half_population
+
+    # Population transitions of interest
+    # If '0.5' appears, it means we are interested in the number of years it took to go from 0.25 -> 0.5
+    population_of_interest = [0.5, 1, 2, 3, 4, 5, 8, 10, 10.4]
```

Comment:
Should this last value be 10.8? It's that in the static chart but perhaps this is revised down in the latest UN WPP?

Reply:
Ah yes I see the max population is 10.4 bil!